src/context/AppProvider/AppContext.tsx

```diff
@@ -93,19 +93,18 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
   useEffect(() => {
     if (!wallet) return
     const walletSupportedChainIds = knownChainIds.filter(chainId => {
-      const chainAccountIds = accountIdsByChainId[chainId] ?? []
-      return walletSupportsChain({ chainId, wallet, isSnapInstalled, chainAccountIds })
+      return walletDeviceSupportsChain({ chainId, wallet, isSnapInstalled })
     })
     dispatch(portfolio.actions.setWalletSupportedChainIds(walletSupportedChainIds))
   }, [accountIdsByChainId, dispatch, isSnapInstalled, wallet])
 
+  // Initial account and portfolio fetch for non-ledger wallets
   useEffect(() => {
-    if (!wallet) return
+    // Skip if wallet is ledger, or if wallet is already connected to accounts - prevents this overriding user selection in account management
+    if (!wallet || isLedger(wallet) || requestedAccountIds.length > 0) return
     ;(async () => {
       let chainIds = Array.from(supportedChains).filter(chainId => {
-        // Note, in this particular case, we are *not* reactive on accountIdsByChainId to avoid extremely costly re-runs of this effect
-        const chainAccountIds = selectAccountIdsByChainIdFilter(store.getState(), { chainId }) ?? []
-        return walletSupportsChain({ chainId, wallet, isSnapInstalled, chainAccountIds })
+        return walletDeviceSupportsChain({ chainId, wallet, isSnapInstalled })
       })
```

Comment:
Also not strictly related to this PR, but given we `dispatch(portfolio.actions.setWalletSupportedChainIds(walletSupportedChainIds))` in the effect above, wondering if that comment (which I wrote for paranoia when I used `store.getState()` here) is still valid, or whether we could potentially use `selectWalletSupportedChainIds` instead?

Assuming `selectWalletSupportedChainIds`'s output is stable enough, (meaning doesn't referentially invalidate on `accountIdsByChainId` change) , we could potentially use that as an initial `chainIds` value here instead.

Reply:
🧠 I like this idea, but will wait for our discussion before changing in case everything changes